source/common/router/config_impl.cc

```diff
@@ -490,9 +490,15 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
 
   if (!route.route().request_mirror_policies().empty()) {
     for (const auto& mirror_policy_config : route.route().request_mirror_policies()) {
-      shadow_policies_.push_back(std::make_unique<ShadowPolicyImpl>(mirror_policy_config));
+      shadow_policies_.push_back(std::make_shared<ShadowPolicyImpl>(mirror_policy_config));
```

Comment:
The motivation of changing pointer ownership style is to avoid deep copy while we're doing `shadow_policies_ = vhost.shadowPolicies();` The alias is already changed to `shared_ptr` as well.

Ideally we might able to alias a ptr to vector, but I suppose we wont't have a too complicated vector in the setting at this moment.